check_gha_pinning/__init__.py

```diff
@@ -37,12 +45,29 @@ def _build_github_url(action: str) -> str:
 
 
 def _get_action_tags(action: str) -> tuple[dict[str, str], dict[str, list[str]]]:
+    """
+    A function to get action tags along with their corresponding SHA values from a given action.
+
+    Args
+    ----
+    action : str
+             the name of the action to get tags for.
+
+    Returns
+    -------
+    by_tag : dict[str, str]
+             a mapping of tag names to their corresponding SHA values.
+    by_sha : dict[str, list[str]]
+             a mapping of SHA values to their corresponding tag names.
+    """
```

Comment:
Nit: I'd go with a less verbose docstring format such as https://google.github.io/styleguide/pyguide.html#383-functions-and-methods
In any case please remove the types from the docstring as we have those in the type annotations already.

Reply:
Meant to remove this - it was just for me getting my head around the return values :D